mytrip/src/main/java/pl/mytrip/trip/Services/QueueJobService.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.stereotype.Service;
+import pl.mytrip.trip.DTOs.Jobs.PosterJobDTO;
+import pl.mytrip.trip.DTOs.Jobs.ThumbnailJobDTO;
+import pl.mytrip.trip.DTOs.Jobs.VideoPresentationDTO;
+import pl.mytrip.trip.Model.Trip;
+import pl.mytrip.trip.Repositories.TripRepository;
+
+import javax.ws.rs.NotFoundException;
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Base64;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class QueueJobService {
+
+    private final TripRepository tripRepository;
+    private static final String JOB_API = "https://tripsbackendprocessingapi.azurewebsites.net";
+
+    public String addThumbnailJob(String photoUrl, String tripId, Long photoId) throws JsonProcessingException {
+        log.info("QueueJobService::addThumbnailJob");
+        ThumbnailJobDTO dto = new ThumbnailJobDTO();
+        dto.setCallbackUrl("/" + tripId + "/photos/" + photoId + "/thumbnail");
+        dto.setFullImageStorageUrl(photoUrl);
+
+        return executePost(JOB_API + "/mediaService/addThumbnailJob",
+                new ObjectMapper().writeValueAsString(dto));
+    }
+
+    public String addPosterJob(String tripId) throws JsonProcessingException {
+        Set<String> photoUrls = new HashSet<>();
+
+        tripRepository.findOne(tripId).getPoints().forEach(waypoint -> waypoint.getPhotos().forEach(photo -> {
+            photoUrls.add(photo.getUrl());
+        }));
+
+        PosterJobDTO dto = new PosterJobDTO();
+        dto.setPhotos(photoUrls);
+        dto.setCallbackUrl("/" + tripId + "/posterCreated");
+
+        return executePost(JOB_API + "/mediaService/addPosterJob",
+                new ObjectMapper().writeValueAsString(dto));
+    }
+
+    public String addVideoPresentationJob(String tripId) throws JsonProcessingException {
+        Set<String> photoUrls = new HashSet<>();
+
+        tripRepository.findOne(tripId).getPoints().forEach(waypoint -> waypoint.getPhotos().forEach(photo -> {
+            photoUrls.add(photo.getUrl());
+        }));
+
+        VideoPresentationDTO dto = new VideoPresentationDTO();
+        dto.setPhotos(photoUrls);
+        dto.setCallbackUrl("/" + tripId + "/presentationCreated");
+
+        return executePost(JOB_API + "/mediaService/addVideoPresentationJob",
+                new ObjectMapper().writeValueAsString(dto));
+    }
+
+    private static String executePost(String targetURL, String urlParameters) {
+        HttpURLConnection connection;
+
+        try {
+            String encoding = Base64.getEncoder().encodeToString(("serviceapp:e&GbE]5as-k~hspC").getBytes());
+            URL url = new URL(targetURL);
+            connection = (HttpURLConnection) url.openConnection();
```

Comment:
jak coś można użyć RestTemplate do zapytań restowych i mapowania odpowiedzi na obiekt, to wtedy można się zmieścić w 2 linijkach :D